src/actions/auth.js

```diff
@@ -0,0 +1,246 @@
+const StrangeAuth = require('strange-auth');
+const WebClient = require('../utils/web-client');
+const History = require('../wiring/history');
+const AuthAct = require('../action-types/auth');
+
+const internals = {};
+
+const actions = exports;
+
+// New User Registration
+
+exports.registrationRequest = () => {
+
+    return {
+        type: AuthAct.REGISTRATION_BEGIN
+    };
+};
+
+exports.registrationSuccess = () => {
+
+    return {
+        type: AuthAct.REGISTRATION_SUCCESS
+    };
+};
+
+exports.registrationFailure = (errMessage) => {
+
+    return {
+        type: AuthAct.REGISTRATION_FAILURE,
+        payload: errMessage
+    };
+};
+
+exports.registerUser = ({ email, password, firstName, lastName }) => {
+
+    return (dispatch) => {
+
+        dispatch(actions.registrationRequest());
+
+        const newUser = WebClient.post('/users', { email, password, firstName, lastName });
+
+        newUser
+
+        .then(({ response }) => {
+
+            dispatch(actions.registrationSuccess());
+            dispatch(actions.login({ email, password }));
+        })
+        .catch((err) => {
+
+            let errMessage = 'Signup failed. Please try again.';
+
+            if (typeof err.response !== 'undefined') {
+                errMessage = err.response.data.message;
+            }
+
+            dispatch(actions.registrationFailure(errMessage));
+        });
+
+        return newUser;
+    };
+
+};
+
+exports.rememberMe = ({ remember }) => {
+
+    return {
+        type: AuthAct.REMEMBER_ME,
+        payload: remember
+    };
+};
+
+// Login and Logout
+
+exports.login = ({ email, password, token }) => {
+
+    return (dispatch) => {
+
+        const strangeLogin = internals.strangeActions.login({ email, password, token });
+
+        return dispatch(strangeLogin)
+
+        .then(() => {
+
+            return History.push('/dashboard');
+
+        })
+        .catch(::console.warn);
```

Comment:
Woah!  This is old.  Is it from an example somewhere?  Would not suggest using the `::` syntax—it was popular for a while, but I don't think it has stuck.